### PR TITLE
pkp/pkp-lib#6567 fix journal section deletion

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/6567

### CODE REVIEW

File: controllers/grid/settings/sections/SectionGridHandler.inc.php [hunk 1]
```diff
@@ -241,15 +241,16 @@ function deleteSection($args, $request) {
 			return new JSONMessage(false, __('manager.setup.errorDeletingItem'));
 		}
 
+		// Validate if it can be deleted
 		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_MANAGER);
 		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
-		$checkSubmissions = $submissionDao->retrieve('SELECT p.publication_id FROM publications p JOIN submissions s ON (s.submission_id = p.submission_id) WHERE p.section_id = ? AND s.context_id = ?', array((int) $request->getUserVar('sectionId'), (int) $journal->getId()));
+		$checkSubmissions = $submissionDao->retrieve('SELECT p.publication_id FROM publications p JOIN submissions s ON (s.submission_id = p.submission_id) WHERE p.section_id = ? AND s.context_id = ?', [(int) $request->getUserVar('sectionId'), (int) $journal->getId()]);
```
Comment (asmecher) [inline controllers/grid/settings/sections/SectionGridHandler.inc.php:247]: The grid handlers shouldn't be executing SQL directly; this should be moved into the DAO.

**Review by asmecher — CHANGES_REQUESTED** [review]

**Comment by bozana** [comment]
@asmecher, I added a new SectionDAO function sectionEmpty and removed the old one that seems not to be used getEmptyByJournalId. OK so?

**Comment by asmecher** [comment]
Looks good, thanks, please go ahead! Feel free to start exploring the Laravel query builder toolset for new queries (https://laravel.com/docs/7.x/queries) -- we'll want to gradually move everything over to using it.

**Comment by asmecher** [comment]
(However, it does look like the tests failed! Worth a look before merging.)

**Pull request merged**